pennylane/devices/qubit/simulate.py

```diff
@@ -131,6 +141,7 @@ def get_final_state(circuit, debugger=None, **execution_kwargs):
             non-negative integer wire labels
         debugger (._Debugger): The debugger to use
         interface (str): The machine learning interface to create the initial state with
+        initial_state (TensorLike): Initial state vector
```

Comment:
Could we instead create fragment tapes with an initial `StatePrep` operation?

Reply:
I thought about it, that and returning `StateMP`. This is how we might implement the generic case, but this would be slow in Lightning (and possibly here) so I kept the prototype as is. I can change it if it is a blocker for you.

Reply:
Let's do this, it will be cleaner.